pages/style-guide/html.md

```diff
@@ -1,77 +1,174 @@
+ - Always use lowercase names for elements and attributes.
+ - Always quote attribute values. Always use double quotes instead of single quotes.
+ - Include the optional closing tags.
+ - Self-closing elements must not be closed.
+ - Optional attributes must be omitted.
+ - `rel`, `type`, `src`, `href` and `class` attributes must be written before any other attribute.
+ - Always include an `alt` attribute for images.
+
+```html
+<!doctype html>
+<html lang="en">
+<head>
+	<meta charset="utf-8">
+	<title>Good Example</title>
+	<link rel="stylesheet" href="style.css">
+	<script src="/jquery.js"></script>
+</head>
+<body>
 
-*   No `type` or `language` attributes on `script` tags.
-*   No `type` attribute on `link` or `style` tags.
+<img class="sample-image" src="assets/img/img.png" alt="Sample Image">
+<p>This is a sample image</p>
 
+</body>
+</html>
 ```
-<script src="..."></script>
-<script></script>
-<link rel="stylesheet" href="...">
-<style></style>
-```
 
-## Indentation
+## HTML Semantics:
+
+Always use HTML elements for what they are meant to be used for.
 
-Don't indent inside `html`, `body`, `script`, or `style`. Indent inside `head` and all other elements.
+## Reducing Markup
 
-### Example
+Always try to avoid superfluous parent elements.
 
+```html
+<!-- Bad HTML -->
+<span class="avatar">
+	<img src="assets/img/img.png" alt="Sample Image">
+</span>
+
+<!-- Good HTML -->
+<img class="avatar" src="assets/img/img.png" alt="Sample Image">
 ```
+
+## Separation of Concerns
+
+Always keep markup, styling and scripting separate.
+
+```html
+<!-- Bad Example -->
 <!doctype html>
 <html lang="en">
 <head>
 	<meta charset="utf-8">
 	<title>Sample Page</title>
+	<script src="script.js"></script>
+</head>
+<body>
 
-	<link rel="stylesheet" href="/style.css">
-	<style>
-	body {
-		font-size: 100em;
-	}
-	</style>
+<h1 style="font-size:1em;line-height:2em;">This is a heading.</h1>
+<p style="text-decoration:underline;font-size:.5em;line-height:1em;">This is an underlined paragraph.</p>
+</body>
+</html>
 
-	<script src="/jquery.js"></script>
-	<script>
-	$(function() {
-		$( "p" ).text( $.fn.jquery );
-	});
-	</script>
+<!-- Good Example -->
+<!doctype html>
+<html lang="en">
+<head>
+	<meta charset="utf-8">
+	<title>Sample Page</title>
+	<link rel="stylesheet" href="style.css">
+	<script src="script.js"></script>
 </head>
 <body>
 
-<p>jQuery is awesome!<p>
-
+<h1>This is a heading.</h1>
+<p>This is an underlined paragraph.</p>
 </body>
 </html>
 ```
+
+## Forms
+
+ - Always use `for` attribute in `label` element. Using `for` attribute in `label` element is more robust, than implicit labelling, across browsers and assistive technology.
+
+```html
+<!-- Bad Example -->
+<label>First<input type="radio" name="input" value="first"></label>
+
+<!-- Good Example -->
+<input type="radio" name="input" id="input-1" value="first">
+<label for="input-1">First</label>
+```
+ - Don't use `placeholder` for labeling.
```

Comment:
Mostly I'm bringing this up because we need to think of how Chassis could give accessible options for users for the labels, but also allow for flexibility in design.